src/control/cmd/daos/util.go

```diff
@@ -165,21 +164,20 @@ func createWriteStream(ctx context.Context, prefix string, printLn func(line str
 			default:
 				line, err := rdr.ReadString('\n')
 				if err != nil {
-					if !(errors.Is(err, io.EOF) || errors.Is(err, os.ErrClosed)) {
+					if !(errors.Is(err, io.EOF)) {
 						printLn(fmt.Sprintf("read err: %s", err))
 					}
 					return
 				}
-				printLn(fmt.Sprintf("%s%s", prefix, line))
+				printLn(line)
 			}
 		}
-	}(ctx, prefix)
+	}(ctx)
 
 	return stream, func() {
-		C.fflush(stream)
 		C.fclose(stream)
-		r.Close()
 		w.Close()
+		done.Lock()
```

Comment:
I'm a bit confused by why you would lock the mutex within a "outCleanup" function. seems like an odd pattern.

Reply:
It is an odd pattern, really in c a semaphore would be better.  I want to block the outCleanup() function from returning until the helper thread has completed draining the pipe which it won't do until it receives EOF.

The overall process is the lock is created and locked, when it comes to cleanup then the cleanup function closes the writer end of the pipe, the worker function keeps reading the pipe until it gets EOF at which point it returns, the lock is unlocked and cleanup completes.

This guarantees the pipe is empty before the context is torn down.